utils/maybe_async/src/lib.rs

```diff
@@ -67,6 +67,131 @@ pub fn maybe_async(_attr: TokenStream, input: TokenStream) -> TokenStream {
     }
 }
 
+/// Conditionally add `async` keyword to functions.
+///
+/// Parses a trait or an `impl` block and conditionally adds the `async` keyword to methods that
+/// are annotated with `#[maybe_async]`, depending on the `async` feature flag being enabled.
+/// Additionally, if applied to a trait definition or impl block, it will add
+/// `#[async_trait::async_trait(?Send)]` to the it.
+///
+/// For example, given the following trait definition:
+/// ```ignore
+/// #[maybe_async_trait]
+/// trait ExampleTrait {
+///     #[maybe_async]
+///     fn hello_world(&self);
+///
+///     fn get_hello(&self) -> String;
+/// }
+/// ```
+///
+/// And the following implementation:
+/// ```ignore
+/// #[maybe_async_trait]
+/// impl ExampleTrait for MyStruct {
+///     #[maybe_async]
+///     fn hello_world(&self) {
+///         // ...
+///     }
+///
+///     fn get_hello(&self) -> String {
+///         // ...
+///     }
+/// }
+/// ```
+///
+/// When the `async` feature is enabled, this will be transformed into:
+/// ```ignore
+/// #[async_trait::async_trait(?Send)]
+/// trait ExampleTrait {
+///     async fn hello_world(&self);
+///
+///     fn get_hello(&self) -> String;
+/// }
+///
+/// #[async_trait::async_trait(?Send)]
+/// impl ExampleTrait for MyStruct {
+///     async fn hello_world(&self) {
+///         // ...
+///     }
+///
+///     fn get_hello(&self) -> String {
+///         // ...
+///     }
+/// }
+/// ```
+///
+/// When the `async` feature is disabled, the code remains unchanged, and neither the `async`
+/// keyword nor the `#[async_trait::async_trait(?Send)]` attribute is applied.
+#[proc_macro_attribute]
+pub fn maybe_async_trait(_attr: TokenStream, input: TokenStream) -> TokenStream {
+    // Try parsing the input as a trait definition
+    if let Ok(trait_item) = syn::parse::<ItemTrait>(input.clone()) {
+        let output = if cfg!(feature = "async") {
+            let mut async_trait = trait_item;
```

Comment:
Question: why do we need to re-define the variable here but we don't need to do it for the impl block on lines 161 - 162 below?

Reply:
I refactored this a bit a couple of times (initially I was outputting both versions but guarded with `#[cfg()]`), and this was a remaining artifact. Just changed it to be similar to the second block.